swap/src/asb/config.rs

```diff
@@ -11,14 +12,46 @@ use serde::{Deserialize, Serialize};
 use std::ffi::OsStr;
 use std::fs;
 use std::path::{Path, PathBuf};
+use std::str::FromStr;
 use tracing::info;
 use url::Url;
 
-const DEFAULT_LISTEN_ADDRESS_TCP: &str = "/ip4/0.0.0.0/tcp/9939";
-const DEFAULT_LISTEN_ADDRESS_WS: &str = "/ip4/0.0.0.0/tcp/9940/ws";
-const DEFAULT_ELECTRUM_RPC_URL: &str = "ssl://electrum.blockstream.info:60002";
-const DEFAULT_MONERO_WALLET_RPC_TESTNET_URL: &str = "http://127.0.0.1:38083/json_rpc";
-const DEFAULT_BITCOIN_CONFIRMATION_TARGET: usize = 3;
+pub trait GetDefaults {
+    fn getConfigFileDefaults() -> Result<Defaults>;
+}
+
+pub struct Defaults {
+    pub default_config_path: PathBuf,
+    default_data_dir: PathBuf,
+    default_listen_address_tcp: Multiaddr,
+    default_listen_address_ws: Multiaddr,
+    default_electrum_rpc_url: Url,
+    default_monero_wallet_rpc_testnet_url: Url,
+    default_bitcoin_confirmation_target: usize,
```

Comment:
Nit: you can probably remove all the leading `default_` part of the variable names.